src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -16,6 +16,7 @@ namespace System.Net.Sockets
     internal static partial class SocketPal
     {
         public const bool SupportsMultipleConnectAttempts = false;
+        public const int MaximumAddressSize = 150; // currently 128 for AF storage on Linux. Adding little bit more for any new protocols
```

Comment:
`128` is enough, there is no need to go higher.
It would be good to add a `static_assert` of this value against `sizeof(sockaddr_storage`) in pal.

Reply:
> `128` is enough, there is no need to go higher.

Linux is always full of surprises :)

Reply:
> Linux is always full of surprises :)

Fortunately we can read the source code :smile:

Reply:
Is sockaddr_storage meant to be the largest AF or is it just currently largest @tmds? If the later, asserting on it would not catch addition of new AF, right?

Reply:
> Is sockaddr_storage meant to be the largest AF or is it just currently largest @tmds? If the later, asserting on it would not catch addition of new AF, right?

sockaddr_storage is meant for storing any AF, so it is is at least as large as the largest AF.
Suppose someone changed the size to account for a new larger AF. That would require re-compiling programs to account for the new size. So this is the type of binary compatibility we try to capture in PAL. Though it requires re-compiling to pick up the new size, existing programs would keep on working as before.